contract/src/main/kotlin/io/provenance/scope/loan/contracts/RecordLoanContract.kt

```diff
@@ -70,11 +72,21 @@ open class RecordLoanContract(
 
     @Function(invokedBy = PartyType.OWNER)
     @Record(LoanScopeFacts.documents)
-    open fun recordDocuments(@Input(LoanScopeFacts.documents) documents: LoanDocuments) = documents.also(documentListInputValidation)
+    open fun recordDocuments(@Input(LoanScopeFacts.documents) documents: LoanDocuments) = documents.also { input ->
+        validateRequirements(VALID_INPUT) {
+            requireThat(
+                input.documentList.isNotEmpty() orError "Must supply at least one document" // TODO: Verify desired; not thrown for optional input
+            )
+            input.documentList.forEach { document ->
+                documentValidation(document)
+            }
+        }
+    }
 
     @Function(invokedBy = PartyType.OWNER)
     @Record(LoanScopeFacts.servicingData)
-    open fun recordServicingData(@Input(LoanScopeFacts.servicingData) servicingData: ServicingData) = servicingData // TODO: Validate input
+    open fun recordServicingData(@Input(LoanScopeFacts.servicingData) servicingData: ServicingData) =
+        updateServicingData(newServicingData = servicingData) // TODO: Add existing record to constructor, annotated as optional?
```

Comment:
Yes, this record will likely be available before this contract is called, so an optional existing record is appropriate. Will have to think about how to combine/reject overwrites.